ESP32 hardware

VIPQualityPost opened the pull request:
we should bring this into upstream so that others can start contributing on the new hardware

dl1com commented on .gitignore:
Please add these entries as well

https://www.toptal.com/developers/gitignore/api/kicad

VIPQualityPost replied:
added the structure and fixed the .gitignore!
git will remove empty folders so they are not reflected in the repo but we can add them back in as we start on layout.

VIPQualityPost replied:
note, archiving the 3d models requires a plugin you can get from the kicad package manager- it is not default functionality of kicad

dl1com commented on ayab-esp32/ayab-esp32.pdf:
Personally, I am not a big fan of committing schematic exports to the repo, as they tend to get outdated easily, as the export step is a manual action and not everybody thinks to do it before committing.
But I understand that it increases accessibility for people who want to have a short overview.

So I would advocate for:
- adding the generation of a schematic export to the GitHub CI, so we get a fresh export on each PR
- only commit exports when doing a release. And when doing so, use GitLFS for it

VIPQualityPost replied:
yes, i'm a strong proponent that there is a .pdf in the repo, otherwise it's annoying to have to clone the repo to see what is going on. 
i'm not sure how to set up the CI - is that on the repo side after you merge? or i need to add this before we bring in the pr?

dl1com replied:
> yes, i'm a strong proponent that there is a .pdf in the repo, otherwise it's annoying to have to clone the repo to see what is going on.

👍 

> i'm not sure how to set up the CI - is that on the repo side after you merge? or i need to add this before we bring in the pr?

As the setup is done file-based (see here: https://github.com/AllYarnsAreBeautiful/ayab-desktop/tree/1.0.0-dev/.github/workflows), we could try already to establish it with this PR.
So we could aready go and do a first shot, inspired by https://github.com/neilenns/KiBot-CICD-Template and its forks and evolve/debug it further. Does not need to be a high end solution in the first shot.
Perhaps @t0mpr1c3  can also support here and there, as he knows GitHub workflow sorcery pretty well.

t0mpr1c3 replied:
Yes, I can take a look.

dl1com reviewed (changes requested):
Would be great if you could add the following folder structure within the ayab-esp32 folder, so we have a common ground right from the beginning.
I love to use this structure, but feel free to improve: https://hackaday.com/2017/05/18/kicad-best-practises-library-management/

dl1com commented:
Hmm... I wonder if we could use this for the project? But it seems it might require a separate repo?
https://github.com/neilenns/KiBot-CICD-Template

dl1com reviewed (changes requested):
- I guess the deletion of the files in the Shieldv2x folder was not intended?
- I'd love to see a README.md which explains some things (we can develop this over time), but initially it should state which KiCAD version is used for this project.
- For my projects, I only commit the symbols, models and footprints which are not contained in the default KiCAD libary to the repo. The default library may change over time, but a copy of the instances is withheld inside the schematic/layout file itself afaik. What's your favorite approach for this, are there best practices?

VIPQualityPost commented:
You're right, I didn't notice, my wildcards were too aggressive. I moved the gitignore template into the ayab-esp32 directory to limit the scope. 

I've added a readme to both the main repo and the new KiCad design.

I agree it would be nice to only include symbols and footprints not from the default library, but it doesn't appear that KiCad can do this (export is *all* symbols and fp), so it would need to be manual. I don't think it's worth the effort and if nothing else, it guarantees the project will always open even if KiCad changes things upstream for the default symbols/footprints (although I doubt that is likely to ever happen). I think we can just keep the exported libraries that include the default symbols/fp. I'm not sure the best practices, I always keep global libraries so I don't have a lot of experience with trying to produce a 'complete package' KiCad project. There is the temp / cache file like you mention, but that seems a little bit like a hack to rely on that for carrying the data instead of a standalone library, unless I misunderstand how it works.

t0mpr1c3 commented:
IMO it doesn't really hurt to export all the symbols.

dl1com commented:
How do we want to proceed from here? Shall we merge it, so we have a common starting point and do further work in form of PRs?

dl1com commented on ayab-esp32/~ayab-esp32.kicad_sch.lck:
I guess this one is not necessary to commit?

VIPQualityPost replied:
I agree, not sure what this is or how it didn't get committed earlier..

VIPQualityPost commented:
I think @jpcornil-git seems happy with most of the decisions... thanks to much of his input!
I would like us to confirm how we are feeling about the user UART, SPI bus and protection on the user buses. Otherwise, I'm happy to merge this and start on digging into component values, choosing parts, footprints, etc.

VIPQualityPost commented:
OK, I think we are good to merge this and start on design finalization/ component selection!
